lib/problem_templates/ttp.ex

```diff
@@ -0,0 +1,56 @@
+defmodule TTP do
+  @moduledoc """
+  Day-scheduling problem template
+  """
+
+  @behaviour Problem
+  alias Types.Chromosome
+  alias Core.Todo
+
+  @impl true
+  @spec genotype(map()) :: Chromosome.t()
+  def genotype(%{todos: todos, time_streak_weights: tsw} = data)
+      when is_map(data) do
+    {_, genes} =
+      Enum.reduce(tsw, {todos, []}, fn
+        streak_weight, {todos_left, chosen_todos} ->
+          result = Todo.find_tasks_within_weight(todos_left, streak_weight)
```

Comment:
Encode todo's weight in chromosome to avoid recomputation of task weight in every fitness evaluation.

Reply:
Not applicable anymore